src/shared/Core.Tests/SettingsTests.cs

```diff
@@ -280,6 +279,78 @@ public void Settings_IsTracingEnabled_EnvarPathy_ReturnsTrueOutValue()
             Assert.Equal(expectedValue, actualValue);
         }
 
+        [Fact]
+        public void Settings_IsTrace2TracingEnabled_EnvarUnset_ReturnsFalse()
+        {
+            var envars = new TestEnvironment();
+            var git = new TestGit();
+
+            var settings = new Settings(envars, git);
+            var result = settings.GetTrace2TracingEnabled(out string actualValue);
+
+            Assert.False(result);
+        }
+
+        [Fact]
+        public void Settings_IsTrace2TracingEnabled_EnvarTruthy_ReturnsTrueOutValue()
+        {
+            const string expectedValue = "1";
+            var envars = new TestEnvironment
+            {
+                Variables = {[Constants.EnvironmentVariables.GcmTrace2] = expectedValue}
+            };
+            var git = new TestGit();
+
+            var settings = new Settings(envars, git);
+            var result = settings.GetTrace2TracingEnabled(out string actualValue);
+
+            Assert.True(result);
+            Assert.Equal(expectedValue, actualValue);
+        }
+
+        [Fact]
+        public void Settings_IsTrace2TracingEnabled_EnvarFalsey_ReturnsFalseOutValue()
+        {
+            const string expectedValue = "0";
+            var envars = new TestEnvironment
+            {
+                Variables = {[Constants.EnvironmentVariables.GcmTrace2] = expectedValue}
+            };
+            var git = new TestGit();
+
+            var settings = new Settings(envars, git);
+            var result = settings.GetTrace2TracingEnabled(out string actualValue);
+
+            Assert.False(result);
+            Assert.Equal(expectedValue, actualValue);
+        }
+
+        [Fact]
+        public void Settings_IsTrace2TracingEnabled_ConfigUnset_ReturnsFalse()
+        {
+            var envars = new TestEnvironment();
+            var git = new TestGit();
+
+            var settings = new Settings(envars, git);
+
+            Assert.False(settings.GetTrace2TracingEnabled(out string actualValue));
+        }
+
+        [Fact]
+        public void Settings_IsTrace2TracingEnabled_ConfigSet_ReturnsTrue()
+        {
+            const string section = Constants.GitConfiguration.Trace2.SectionName;
+            const string property = Constants.GitConfiguration.Trace2.EventTarget;
+
+            var envars = new TestEnvironment();
+            var git = new TestGit();
+            git.Configuration.Global[$"{section}.{property}"] = new[] {"af_unix:./test.uds"};
```

Comment:
the "af_unix:" prefix is Linux/MacOS specific.
Windows will have a "//./pipe/" prefix (and/or the backslash version of that string).